staging/src/k8s.io/apiextensions-apiserver/test/integration/ratcheting_test.go

```diff
@@ -164,7 +165,12 @@ func (a applyPatchOperation) Do(ctx *ratchetingTestContext) error {
 
 	patch := &unstructured.Unstructured{}
 	if obj, ok := a.patch.(map[string]interface{}); ok {
-		patch.Object = obj
+		patch.Object = map[string]interface{}{}
+
+		// Copy the map at the top level to avoid modifying the original.
+		for k, v := range obj {
+			patch.Object[k] = v
+		}
```

Comment:
remind me why we need this change now?

Reply:
The map is coming in from the test case.

We first run the test case against spec, and then status.

When we run it against spec, we (below) write some extra fields to it (apiVersion, kind, metadata), which updates the map within the test case.

When we then run it against status, those fields are carried over from the spec test, and end up as status fields (status.apiVersion, status.kind, status.metadata), which then get rejected as unknown fields by the API

By coping the data out of the map, we prevent this pollution between the spec and status test case

Reply:
Ack. Makes sense. There is `runtime.DeepCopyJSON` for that purpose. Would clarify and simplify what we are doing here.

Reply:
I tried using a DeepCopy already, but any deep copy combination I end up with `panic: cannot deep copy int`, so I think we need to leave this as is.

The alternative I guess is to serialise and deserialise the string to create a copy, is that simpler than just copying the map out?

Reply:
you have to type cast the ints in the test cases to int64. Technically, these are invalid now.